fastlane/lib/fastlane/actions/carthage.rb

```diff
@@ -86,7 +91,16 @@ def self.available_options
                                        description: "Define which platform to build for",
                                        optional: true,
                                        verify_block: proc do |value|
-                                         UI.user_error!("Please pass a valid platform. Use one of the following: all, iOS, Mac, watchOS") unless ["all", "iOS", "Mac", "watchOS"].include? value
+                                         value.split(',').each do |platform|
+                                           UI.user_error!("Please pass a valid platform. Use one of the following: #{available_platforms.join(', ')}") unless available_platforms.include? platform
+                                         end
+                                       end),
+          FastlaneCore::ConfigItem.new(key: :configuration,
+                                       env_name: "FL_CARTHAGE_CONFIGURATION",
+                                       description: "Define which build configuration to use when building",
+                                       optional: true,
+                                       verify_block: proc do |value|
+                                         UI.user_error!("Please pass a valid build configuration. You can review the list of configurations for this project using the following command: `xcodebuild -list`") if value.nil? or value.chomp(' ').empty?
```

Comment:
I'm pretty sure the order of the checks with the `or` is wrong here. Please switch to a multi-line statement instead, since the line is very long here